### PR TITLE
feat(x2 mobile): doubles the counted data for mobile uploads PE-2028

### PR DESCRIPTION
**Pull request by matibat**

### CODE REVIEW

File: src/daily_output.ts [hunk 1]
```diff
@@ -335,6 +342,25 @@ export class DailyOutput {
 		this.data.blockHeight = currentHeight;
 	};
 
+	private doublesData(txDate: Date, appName?: string, appPlatform?: string): boolean {
+		const MOBILE_x2_START_DATE = new Date('10/30/2022');
+		const MOBILE_x2_END_DATE = new Date('11/27/2022');
+		const daysDiffToStart = daysDiffInEST(txDate, MOBILE_x2_START_DATE);
+		const daysDiffToEnd = daysDiffInEST(txDate, MOBILE_x2_END_DATE);
+		const doublingAlreadyStarted = daysDiffToStart === 0;
+		const doublingDidntFinish = daysDiffToEnd <= 7 * 4;
```
Comment (matibat) [inline src/daily_output.ts]: Works just perfectly for the dates BEFORE and WHILE happening, but AFTER the event it's still doubling the mobile uploads' data count. 

TODO: better determine when the event finishes

Comment (matibat) [reply src/daily_output.ts]: [fc6d906](https://github.com/ardriveapp/inferno-backend/pull/56/commits/fc6d906e9c2762d9bf103c8de848e17e20f79194)

File: src/daily_output.ts [hunk 2]
```diff
@@ -335,6 +343,26 @@ export class DailyOutput {
 		this.data.blockHeight = currentHeight;
 	};
 
+	private doublesData(txDate: Date, appName?: string, appPlatform?: string): boolean {
+		const MOBILE_x2_START_DATE = new Date('10/30/2022'); // Oct 30th, 2022
+		const MOBILE_x2_END_DATE = new Date('11/27/2022'); // Nov 27th, 2022
+
+		// set it to the day after because isDateInRangeOfDSays expects the end to be exluded from the range
+		MOBILE_x2_END_DATE.setTime(MOBILE_x2_END_DATE.getTime() + 1000 * 60 * 60 * 24);
+
```
Comment (matibat) [inline src/daily_output.ts]: November 27th is not included
```suggestion
```

File: src/common.ts [hunk 3]
```diff
@@ -243,16 +243,34 @@ export function isSemanticVersionGreaterThan(appVersion_a: string, appVersion_b:
 	return false;
 }
 
+/**
+ * Returns true if the date is after the day_start and before the day_end
+ * @param day_start the day when the range starts
+ * @param day_end the day when the range ends (not included)
+ * @param date the date to check
+ */
+export function isDateInRangeOfDays(day_start: Date, day_end: Date, date: Date): boolean {
```
Comment (karlprieb) [inline src/common.ts:252]: Can we have tests for `isDateInRangeOfDays`?

Comment (matibat) [reply src/common.ts:252]: There we have
ec6596fcc8b4c532efd956aed5b1941f7b1ecb12

**Review by karlprieb — APPROVED** [review]

**Pull request merged**